src/lib/metrics.ts

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+
+import promClient from "prom-client";
+import { performance } from "perf_hooks";
+
+export class MetricsCollector {
+  private registery: promClient.Registry;
+  private errors: promClient.Counter<string>;
```

Comment:
ironically, everywhere but the `controller.ts`, we are using the same convention of prefixing the private fields with an `_`,:

`private _errors: promClient.Counter<string>;`